controllers/pulp/api.go

```diff
@@ -1021,7 +1021,8 @@ func serviceAPISpec(name, namespace, deployment_type string) corev1.ServiceSpec
 			"app":                          "pulp-api",
 			"pulp_cr":                      name,
 		},
-		SessionAffinity: serviceAffinity,
-		Type:            serviceType,
+		SessionAffinity:          serviceAffinity,
+		Type:                     serviceType,
+		PublishNotReadyAddresses: true,
```

Comment:
When the operator is provisioning the components, `pulp-web` pods keep crashing until `pulp-api`/`pulp-content` pods get into a `READY` state.
Checking `pulp-web` pod logs they are failing to resolve the `Service` hostnames:
```
nginx: [emerg] host not found in upstream "example-pulp-content-svc:24816" in /etc/opt/rh/rh-nginx116/nginx/nginx.conf:21
```
this causes the `nginx` process to crash and, as `nginx` is the main process of the container, the container itself (and finally the pod) will also crash.


I believe that this is happening because of a feature of `kube-dns` that does not propagate the hostname while the pods are in a not `READY` state:
```
publishNotReadyAddresses  <boolean>
     publishNotReadyAddresses indicates that any agent which deals with
     endpoints for this Service should disregard any indications of
     ready/not-ready. The primary use case for setting this field is for a
     StatefulSet's Headless Service to propagate SRV DNS records for its Pods
     for the purpose of peer discovery. The Kubernetes controllers that generate
     Endpoints and EndpointSlice resources for Services interpret this to mean
     that all endpoints are considered "ready" even if the Pods themselves are
     not. Agents which consume only Kubernetes generated endpoints through the
     Endpoints or EndpointSlice resources can safely assume this behavior.
```

After setting `publishNotReadyAddresses=true` `pulp-web` pods stopped crashing.